docs/decisions/0045-updated-vector-store-design.md

```diff
@@ -0,0 +1,810 @@
+class AzureAISearchMLIndexMemoryService(AzureAISearchMemoryCollectionUpdateService updateService): MemoryCollectionService(updateService);
+
+// Customer collections service implementation, that uses the base Azure AI Search implementation for get, doesExist and delete, but adds it's own creation.
+class ContosoProductsMemoryCollectionService(AzureAISearchMemoryCollectionUpdateService updateService): MemoryCollectionService(updateService);
+
+```
+
+#### Option 5 - Separated collection and record management with collection create separate from other operations, with overall aggregation class on top.
+
+Same as option 3 / 4, plus:
+
+```cs
+
+interface IMemoryService : IMemoryCollectionCreateService, IMemoryCollectionService, IMemoryRecordService
+{    
+}
+
+// Create a static factory that produces one of these, so only the interface is public, not the class.
+internal class CombinedMemoryService<TDataModel>(IMemoryCollectionCreateService creation, IMemoryCollectionService collections, IMemoryRecordService<TDataModel> records): IMemoryService
+{
+}
+
+```
+
+#### Decision Outcome
+
+Chosen option: 4 + 5.
+
+- Collection create and configuration varies considerably across different schemas, even for the same database, but also varies by database type.
+- Collection list, exists and delete is the same across different schemas, but varies by database type.
+- Vector storage, even with custom schemas can be supported using a single implementation per database type.
+- We will therefore need to support multiple collection create service implementations per store type, a single collection update service implementation per store type, and a single vector store implementation per store type.
+- At the same time we can layer interfaces on top that allow easy combined access to collection and record management.
+
+
+###  Question 2: Collection name and key value normalization in store, decorator or via injection.
+
+#### Option 1 - Normalization in main record service
+
+- Pros: Simple
+- Cons: The normalization needs to vary separately from the record service, so this will not work
+
+```cs
+    public class AzureAISearchMemoryRecordService<TDataModel> : IMemoryRecordService<TDataModel>
+    {
+        ...
+
+        // On input.
+        var normalizedCollectionName = this.NormalizeCollectionName(collectionName);
+        var encodedId = AzureAISearchMemoryRecord.EncodeId(key);
+
+        ...
+
+        // On output.
+        DecodeId(this.Id)
+
+        ...
+    }
+```
+
+#### Option 2 - Normalization in decorator
+
+- Pros: Allows normalization to vary separately from the record service.
+- Pros: No code executed when no normalization required.
+- Pros: Easy to package matching encoders/decoders together.
+- Pros: Easier to obsolete encoding/normalization as a concept.
+- Cons: Need to implement the full MemoryRecordService interface.
```

Comment:
Is this really a Con based on our current requirements or should this be Neutral? 

We can introduce an `IReadOnlyMemoryRecordService` later if needed?

Reply:
Updating to clarify.  This isn't a major con, but I'm just contrasting it with option 3, where we can just pass the translation functions and we don't need to do a full wrapper class.